harvester/file.py

```diff
@@ -174,6 +174,44 @@ def download(self, base_path=None, file_dir=None, filename=None):
             source.raise_for_status()
             output_file.write(source.content)
 
+    def download_to_remote(
```

Comment:
The current implementation here makes some compromises I think we can improve on. I think that the ideal option would be to either
1. save the file directly to Puhti, no intermediate storage on the airflow machine
- pro: no need to write the file to disk twice
- pro: less room for introducing errors into the file when reading/writing
- con: the transfer from local -> puhti and the possible complications are intermixed with the overall download logic
2. keep harvester simple and local, let Airflow take care of transfers
- pro: Airflow has a ready-made  provider for this (I think?)
- pro: Easy to switch the transfer technology (sftp to Puhti vs s3 to Allas vs something else) without touching the basic file download
- con: two transfers, i.e. double the transfer time and double the chance for corruption

But now we both write the file to disk on two locations and have Puhti access as a responsibility of a file object. Does it sound like a realistic/good idea to transition to one of the scenarios above to benefit from at least some of the pros listed?